platforms/core-configuration/configuration-cache/src/test/kotlin/org/gradle/internal/cc/impl/serialization/codecs/BindingsBackedCodecTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl.serialization.codecs
+
+import org.gradle.internal.cc.base.exceptions.ConfigurationCacheError
+import org.gradle.internal.serialize.BaseSerializerFactory.BOOLEAN_SERIALIZER
+import org.gradle.internal.serialize.BaseSerializerFactory.INTEGER_SERIALIZER
+import org.gradle.internal.serialize.BaseSerializerFactory.STRING_SERIALIZER
+import org.gradle.internal.serialize.graph.codecs.Bindings
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.containsString
+import org.junit.Assert.fail
+import org.junit.Test
+
+class BindingsBackedCodecTest : AbstractUserTypeCodecTest() {
+    @Test
+    fun `throws if collection reports invalid size`() {
```

Comment:
💭It makes sense if you know the implementation, but it's hard to connect the name of the test with its body, since nothing points to collection size, and instead different bindings are used for write and read 🤔
I understand, it's to make the tag of the string serializer to be different, but then again, this test seems to rely on knowing the implementation

Reply:
> 💭It makes sense if you know the implementation, but it's hard to connect the name of the test with its body, 

This is actually a copy-paste mistake 🙃